zone/task_manager.cpp

```diff
@@ -1814,3 +1814,96 @@ void TaskManager::SyncClientSharedTaskStateToLocal(
 		}
 	}
 }
+
+void TaskManager::HandleUpdateTasksOnKill(Client *client, uint32 npc_type_id)
+{
+	for (auto &c: client->GetPartyMembers()) {
+		if (!c->ClientDataLoaded() || !c->HasTaskState()) {
+			continue;
+		}
+
+		LogTasksDetail("[HandleUpdateTasksOnKill] Looping through client [{}]", c->GetCleanName());
+
+		// loop over the union of tasks and quests
+		for (auto &active_task : c->GetTaskState()->m_active_tasks) {
+			auto current_task = &active_task;
+			if (current_task->task_id == TASKSLOTEMPTY) {
+				continue;
+			}
+
+			// Check if there are any active kill activities for this p_task_data
+			auto p_task_data = m_task_data[current_task->task_id];
+			if (p_task_data == nullptr) {
+				return;
+			}
+
+			for (int activity_id = 0; activity_id < p_task_data->activity_count; activity_id++) {
+				ClientActivityInformation *client_activity = &current_task->activity[activity_id];
```

Comment:
We should really combine all the update methods that re-use these activity checks at some point, I think they only differ in specifying the TaskActivityType and maybe some goal stuff, but it's duplicated across like 6 methods

Reply:
It was actually substantially worse at one point. There's a lot I'd like to change about the legacy task system structure